### PR TITLE
Scala 3 Support

### PR DESCRIPTION
**Pull request by raquo**

### CODE REVIEW

File: shared/src/main/scala-3.0.0-RC2/urldsl/vocabulary/FromStringWithNumeric.scala [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package urldsl.vocabulary
+
+import urldsl.errors.ErrorFromThrowable
+
+trait FromStringWithNumeric {
```
Comment (raquo) [inline shared/src/main/scala-3.0.0-RC2/urldsl/vocabulary/FromStringWithNumeric.scala:5]: I'm not sure what this should be for Scala 3. This trait is empty in 2.12, I copied this from 2.13

File: shared/src/test/scala/urldsl/examples/PathSegmentExamples.scala [hunk 2]
```diff
@@ -60,13 +60,13 @@ final class PathSegmentExamples extends AnyFlatSpec with Matchers {
     /** You can also "group" several segments into a more meaningful class than a pair or a triplet. */
     case class Stuff(str: String, j: Int)
 
-    (s1 / s2).as((Stuff.apply _).tupled, Stuff.unapply(_: Stuff).get).matchRawUrl(sampleUrl) should be(
+    (s1 / s2).as((t: (String, Int)) => Stuff(t._1, t._2), (s: Stuff) => s match { case Stuff(str, j) => (str, j) }).matchRawUrl(sampleUrl) should be(
```
Comment (raquo) [inline shared/src/test/scala/urldsl/examples/PathSegmentExamples.scala:63]: I couldn't find a more elegant way to satisfy both Scala versions.

`unapply` [changing signature](https://scalacenter.github.io/scala-3-migration-guide/docs/incompatibilities/other-changed-features.html)

File: shared/src/test/scala/urldsl/language/PathSegmentRawUrlSpec.scala [hunk 3]
```diff
@@ -29,7 +29,7 @@ final class PathSegmentRawUrlSpec extends AnyFlatSpec with Matchers {
 
     val ls = List("car", "motorcycle")
 
-    (root / segment[String] / "bold" / remainingSegments).path(hello, ls) should be(
+    (root / segment[String] / "bold" / remainingSegments).path((hello, ls)) should be(
```
Comment (raquo) [inline shared/src/test/scala/urldsl/language/PathSegmentRawUrlSpec.scala:32]: Note, Scala 3 doesn't do auto-tupling here.

**Comment by sherpal** [comment]
Thanks for this! I can bump this to be 0.4.0 and publish as is for 2.12, 2.13 and 3-RC2.

**Pull request merged**